### DISCUSSION TITLE
Unable to Build an Azure ML Container Image for an MLflow model

### DISCUSSION THREAD

**Discussion by manishpyy** [mlflow/mlflow#5578] [Q&A]
I am unable to build an Azure ML Container Image for an MLflow model. While building the docker image using the code:

azure_image, azure_model = mlflow.azureml.build_image(model_uri="<model_uri path>",
                                                      workspace=<azure_workspace>,
                                                      synchronous=True)

Getting the following error in the log:
Step 9/13 : RUN pip install mlflow==1.24.0
 ---> Running in c52dae9d410a
[91mERROR: Could not find a version that satisfies the requirement mlflow==1.24.0 (from versions: 0.0.1, 0.1.0, 0.2.0, 0.2.1, 0.3.0, 0.4.0, 0.4.1, 0.4.2, 0.5.0, 0.5.1, 0.5.2, 0.6.0, 0.7.0, 0.8.0, 0.8.1, 0.8.2, 0.9.0, 0.9.0.1, 0.9.1, 1.0.0, 1.1.0, 1.1.1.dev0, 1.2.0, 1.3.0, 1.4.0, 1.5.0, 1.6.0, 1.7.0, 1.7.1, 1.7.2, 1.8.0, 1.9.0, 1.9.1, 1.10.0, 1.11.0, 1.12.0, 1.12.1, 1.13, 1.13.1, 1.14.0, 1.14.1, 1.15.0, 1.16.0, 1.17.0, 1.18.0, 1.19.0, 1.20.0, 1.20.1, 1.20.2, 1.21.0, 1.22.0, 1.23.0, 1.23.1)
[0m[91mERROR: No matching distribution found for mlflow==1.24.0

**Comment by eeshsaxena** [comment]
Based on publicly available information about `mlflow`:

For SWE-bench style coding benchmarks, the results typically vary based on the model, agent scaffolding, and the specific benchmark subset (verified vs full).

Top performing models as of mid-2026 for software engineering tasks:
- **Claude Sonnet 3.5 / 3.7** - generally leads on SWE-bench verified
- **GPT-4o** - strong baseline, slightly behind top Claude models on coding tasks
- **Gemini 1.5 Pro / 2.0** - competitive on code generation
- **DeepSeek-V3** - strong open-weight option for coding

For local/open-source models that work well with tool-calling (needed for agentic tasks like SWE-bench):
- **Qwen2.5-Coder-32B** - best open model for code as of late 2025
- **DeepSeek-Coder-V2** - strong alternative
- **Llama-3.1-70B** - good general purpose with tool-calling support

The specific 48% result would depend on which model version and prompting strategy was used. Check the repo's benchmark docs or recent blog posts for the exact setup.
